src/app/components/quick-connect-dialog/quick-connect-dialog.component.ts

```diff
@@ -160,14 +153,16 @@ export class QuickConnectDialogComponent implements OnInit, OnDestroy {
         zip(suggestions$, existingChannels$)
             .pipe(
                 map(([suggestions, channels]) =>
-                    suggestions.filter(
-                        (suggestion) =>
-                            !channels.find(
-                                (channel) =>
-                                    channel.partner_address ===
-                                    suggestion.address
-                            )
-                    )
+                    suggestions.filter((suggestion) => {
+                        const channelExisting = !!channels.find(
+                            (channel) =>
+                                channel.partner_address === suggestion.address
+                        );
+                        const isOwnAddress =
+                            suggestion.address ===
+                            this.raidenService.raidenAddress;
+                        return !channelExisting && !isOwnAddress;
```

Comment:
No comment but compliment: I like that you use descriptive names for the boolean expressions here. This makes the code very clean and super easy to read like a book. Good job! 👍🏾 
(just close the conversation)

Reply:
That's also precious feedback, thank you :smiley: